utils/include/detray/simulation/event_generator/random_track_generator.hpp

```diff
@@ -302,10 +329,11 @@ class random_track_generator
     constexpr random_track_generator() = default;
 
     /// Construct from external configuration
+    /// If a concrete seed was given, use it otherwise prefer random seeds
     DETRAY_HOST_DEVICE
-    constexpr random_track_generator(configuration cfg) : m_gen{}, m_cfg(cfg) {
-        m_gen.engine.seed(m_cfg.seed());
-    }
+    constexpr random_track_generator(configuration cfg)
+        : m_gen{(cfg.seed() != 0u) ? generator_t{cfg.seed()} : generator_t{}},
```

Comment:
This can be simplified as `m_gen(generator_t{cfg.seed()})`

Reply:
Then there is no way of triggering the random seeding?

Reply:
Do we need a random seeding?

Reply:
We still don't lose randomness if we set different seed in configurator.

Reply:
Does it harm us to have it? I would argue it might produce better random numbers

Reply:
random_generator is already random - why would we have two layers of random?